polyglot.js

```diff
@@ -71,29 +104,58 @@ class PolyGlot {
 			actors.push(game.user.character);
 		for (let actor of actors) {
 			try {
-				// Don't duplicate the value in case it's a not an array
-				for (let lang of actor.data.data.traits.languages.value)
-				    this.known_languages.add(lang)
-			} catch (err) { 
+				switch(game.system.id.toLowerCase()){
+					default:
+						// Don't duplicate the value in case it's a not an array
+						for (let lang of actor.data.data.traits.languages.value)
+								this.known_languages.add(lang)
+						break;
+					case "wfrp4e":
+						let re = /Language \((.+)\)/i;
+						for (let item of actor.data.items){
+							let found = item.name.match(re);
+							if(found) {
+								// adding only the descriptive language name, not "Language (XYZ)"
+								this.known_languages.add(found[1]);
+							}
+						}
+						break;
+				}
+			} catch (err) {
 				// Maybe not dnd5e, pf1 or pf2e or corrupted actor data?
 			}
 		}
 		if (this.known_languages.size == 0) {
 			if (game.user.isGM)
 				this.known_languages = new Set(Object.keys(PolyGlot.languages))
-			else
-				this.known_languages.add("common");
+			else {
```

Comment:
Coding style: If adding an else which uses curly braces, then the one liner `if` statement should also become enclosed in curly braces